### ISSUE TITLE
fix: randperm dtype=None and batch_norm 1-D input robustness

### ISSUE THREAD

**Issue by Postroggy** [FlagOpen/FlagGems#5720]
## Issue Type
- [x] Bug Report

## Summary

Two robustness bugs (measured on RTX 5060 via the torch_fl flag_gems route,
flag_gems 7fb49bad):

1. **randperm** asserts `dtype` is int16/int32/int64, but `dtype=None` (the
   PyTorch default, which resolves to int64) fails the assert with an empty
   `AssertionError`. ATen callers that pass `dtype=None` explicitly (as the
   torch_fl overload survey synthesizer does) hit this; `torch.randperm`
   works only because the Python layer defaults dtype before dispatch.
2. **batch_norm** unpacks the reshaped input with
   `batch_dim, feat_dim, spatial_dim = input_3d.shape`; a 1-D input
   (invalid for batch norm, but PyTorch's CPU reference rejects it with a
   clear error) makes the unpack fail inside the kernel launch path, which
   the torch_fl survey records as a process crash rather than a clean error.

## Reproduction

```python
import torch
import flag_gems

torch.ops.aten.randperm.default(32, dtype=None, layout=None,
                                device=torch.device("cuda"), pin_memory=False)
# -> AssertionError (empty message)

torch.ops.aten.native_batch_norm(torch.randn(32, device="cuda"),
                                 torch.randn(1, device="cuda"),
                                 torch.randn(1, device="cuda"),
                                 None, None, True, 0.1, 1e-5)
# -> process crash (unpack error in launch path)
```

## Expected vs Actual

**Expected:** randperm(dtype=None) returns an int64 permutation; batch_norm
with a 1-D input raises a clear error like PyTorch's.

**Actual:** randperm raises an empty AssertionError; batch_norm crashes the
process.

## Proposed Fix

- `randperm`: default `dtype = torch.int64` when None before the assert.
- `batch_norm`: reject `input.dim() < 2` with a clear ValueError before
  reshaping.

## Verification Plan
- [x] randperm(dtype=None): OK, int64 output
- [x] batch_norm 1-D: ValueError (clean), 4-D unchanged
